### ISSUE TITLE
Error Pages not working (Handler.ts)

### ISSUE THREAD

**Issue by claudesortwell** [adonisjs/core#3414]
## What is happening?
Adonis is not rendering error pages defined in statusPages in handler.ts, this also does not working in dev using the optional boolean. A prime example is the [Adonis Poll example](https://polls.adonisjs.dev/qwd), but even in my private repo, I have the same problem.

## Package version
```
 "devDependencies": {
    "@adonisjs/assembler": "^5.3.0",
    "@symfony/webpack-encore": "^1.7.0",
    "adonis-preset-ts": "^2.1.0",
  },
  "dependencies": {
    "@adonisjs/core": "^5.4.0",
    "@adonisjs/drive-s3": "^1.0.5",
    "@adonisjs/mail": "^7.2.4",
    "@adonisjs/repl": "^3.1.0",
    "@adonisjs/session": "^6.1.2",
    "@adonisjs/view": "^6.1.1",
}
```

## Node.js and npm version
v14.16.1, Yarn 3.1.1

## Sample Code (to reproduce the issue)
```
  protected statusPages = {
    '403': 'pages/errors/unauthorized',
    '404': 'pages/errors/not-found',
    '500..599': 'pages/errors/server-error',
  }
```

## BONUS (a sample repo to reproduce the issue)
[Polls App Example route not founds not working](https://github.com/adonisjs-community/polls-app )
https://polls.adonisjs.dev/qwfqwfeqf. According to the repos handler.ts, it should render the route not found edge page but the link is rendering the default error: 
```
E_ROUTE_NOT_FOUND: Cannot GET:/qwfqwfeqf
```

**Comment by thetutlage** [comment]
There was an issue with the polls app in which the exception handler itself was raising exception. Hence the server returned a generic error. I have fixed that error here https://github.com/adonisjs-community/polls-app/commit/553f25d6decfe88cf72ada4af33c58b48e41fc80 and now the poll app does render status pages.

Are you testing the same on local? If yes, then status pages are only rendered when the `NODE_ENV=production`. However, there is a flag to force status page to render in development too. https://docs.adonisjs.com/guides/exception-handling#status-pages

**Comment by claudesortwell** [comment]
I'll provide a sample repo tomorrow for you. Ive tried adding the force status flag, no luck. I think it might have to do with starting adonis with the api template then moving adding web.

**Comment by thetutlage** [comment]
Ohh is it that you are using the API template? If yes, then shouldn't you expect your API server to always return JSON and not HTML?

**Comment by claudesortwell** [comment]
Yeah ive used the API Template then added Views. Guessing there is some other thing im supposed to tweak

**Comment by thetutlage** [comment]
Yes, as of now the status pages are not for the API template. However, I am still curious to know why you want to return HTML from an API server?

**Comment by claudesortwell** [comment]
Yeah makes sense. It's kinda of a hybrid application. So API for customer's SPA then we have an admin portal built with the adonis views.

You can probably close this now, as I understand your reasoning for not adding views in an API template. Maybe adding some middleware error catching or a custom handler will work for hybrid applications.

**Comment by thetutlage** [comment]
You can still manage it if your API is nested within some prefix like `/api`. Then you can use the changes implemented in this PR https://github.com/adonisjs/http-server/pull/51/files#diff-02b5b8ba60024cffe8cebc1ff41db79cd8baab6e1fb61101ffbcd2eff3e04ff3R591

Lemme know if that is the case and you need any help with implementation
